core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java

```diff
@@ -276,38 +305,22 @@ protected boolean isBusy(@NonNull Node node, @NonNull Session session) {
   protected boolean isResponseRateInsufficient(@NonNull Node node, long now) {
     // response rate is considered insufficient when less than 2 responses were obtained in
     // the past interval delimited by RESPONSE_COUNT_RESET_INTERVAL_NANOS.
-    if (responseTimes.containsKey(node)) {
-      AtomicLongArray array = responseTimes.get(node);
-      if (array.length() == 2) {
-        long threshold = now - RESPONSE_COUNT_RESET_INTERVAL_NANOS;
-        long leastRecent = array.get(0);
-        return leastRecent - threshold < 0;
-      }
+    AtomicLongArray array;
+    try {
+      array = responseTimes.asMap().containsKey(node) ? responseTimes.get(node) : null;
+    } catch (ExecutionException e) {
+      throw new RuntimeException(e);
     }
-    return true;
+    if (array == null) return true;
+    else if (array.length() == 2) {
+      long threshold = now - RESPONSE_COUNT_RESET_INTERVAL_NANOS;
+      long leastRecent = array.get(0);
+      return leastRecent - threshold < 0;
+    } else return true;
   }
 
   protected void updateResponseTimes(@NonNull Node node) {
-    responseTimes.compute(
-        node,
-        (n, array) -> {
-          // The array stores at most two timestamps, since we don't need more;
-          // the first one is always the least recent one, and hence the one to inspect.
-          long now = nanoTime();
-          if (array == null) {
-            array = new AtomicLongArray(1);
-            array.set(0, now);
-          } else if (array.length() == 1) {
-            long previous = array.get(0);
-            array = new AtomicLongArray(2);
-            array.set(0, previous);
-            array.set(1, now);
-          } else {
-            array.set(0, array.get(1));
-            array.set(1, now);
-          }
-          return array;
-        });
+    responseTimes.refresh(node);
```

Comment:
maybe add some comments about why this will reload the key synchronously (LoadingCache#reload says this may be async but i don't think it is here)